packages/analytics-core/src/core-client.ts

```diff
@@ -287,13 +286,94 @@ export class AmplitudeCore implements CoreClient {
     return this.process(event);
   }
 
+  /**
+   *
+   * This method applies identify operations to user properties and
+   * returns a single object representing the final user property state.
+   *
+   * This is a best-effort api that only supports $set, $clearAll, and $unset.
+   * Other operations are not supported and are ignored.
+   *
+   *
+   * @param userProperties The `event.userProperties` object from an Identify event.
+   * @returns A key-value object user properties without operations.
+   *
+   * @example
+   * Input:
+   * {
+   *   $set: { plan: 'premium' },
+   *   custom_flag: true
+   * }
+   *
+   * Output:
+   * {
+   *   plan: 'premium',
+   *   custom_flag: true
+   * }
+   */
+  getOperationAppliedUserProperties(userProperties: UserProperties | undefined): { [key: string]: any } {
+    const updatedProperties: { [key: string]: any } = {};
+
+    if (userProperties === undefined) {
+      return updatedProperties;
+    }
+
+    // Keep non-operation keys for later merge
+    const nonOpProperties: Record<string, any> = {};
+    for (const key in userProperties) {
+      if (!Object.values(IdentifyOperation).includes(key as IdentifyOperation)) {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        nonOpProperties[key] = userProperties[key];
+      }
+    }
+
+    OrderedIdentifyOperations.forEach((operation) => {
+      if (!Object.keys(userProperties).includes(operation)) return;
+
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+      const opProperties = userProperties[operation];
+
+      switch (operation) {
+        case IdentifyOperation.CLEAR_ALL:
+          for (const prop in updatedProperties) {
+            // Due to operation order, the following line will never execute.
+            /* istanbul ignore next */
+            delete updatedProperties[prop];
+          }
+          break;
+        case IdentifyOperation.UNSET:
+          for (const prop in opProperties) {
+            delete updatedProperties[prop];
+          }
+          break;
+        case IdentifyOperation.SET:
+          Object.assign(updatedProperties, opProperties);
+          break;
+      }
+    });
+
+    // Merge non-operation properties.
+    // Custom properties should not be affected by operations.
+    // https://github.com/amplitude/nova/blob/343f678ded83c032e83b189796b3c2be161b48f5/src/main/java/com/amplitude/userproperty/model/ModifyUserPropertiesIdent.java#L79-L83
+    Object.assign(updatedProperties, nonOpProperties);
```

Comment:
this is additional behavior from what we had discussed earlier?

Reply:
Yes. I add this because the user properties type allows custom properties (no operations) https://github.com/amplitude/Amplitude-TypeScript/blob/43441594b6dd5e23bb09159a5bac8294a8948034/packages/analytics-core/src/types/event/event.ts#L88-L92.

Reply:
ok, thanks - will add to iOS as well